File: ejercicio5.java
import java.io.*;
import java.util.*;

class Main
{
    int x=0;
    
    public void fn()
    {
        x = 21;
        System.out.println("Variable global "+x);
        int x;
        x = 42;
        System.out.println("Variable local "+x);
    }
    
    public static void main (String[] args) 
    {
        fn();
        System.out.println("Variable global "+x);
    }
}

File: ejercicio5.cpp
#include <stdlib.h>
#include <stdio.h>
#include <iostream>

using namespace std;

int x;

void fn()
{
    x = 21;
    cout<<"Variable global"<<x<<endl;
    int x;
    x = 42;
    cout<<"Variable global"<<x<<endl;
}

int main()
{
    fn();
    cout<<"Variable global"<<x<<endl;
    
    return 0;
}